### ISSUE TITLE
SQLSTATE[22007]: Invalid datetime format

### ISSUE THREAD

**Issue by jgacuca567** [GoogleCloudPlatform/getting-started-php#173]
@ryanmats @bshaffer @tmatsuo 

Following Issue https://github.com/GoogleCloudPlatform/php-docs-samples/issues/439 

I was able to get to the next step. However the datetime issue is popping up. 

When I clicked Add New Book, The Date is shown by Default

<img width="1227" alt="screen shot 2017-09-16 at 11 45 49 am" src="https://user-images.githubusercontent.com/4000076/30513642-9e6a0986-9ad4-11e7-9861-842251f8527a.png">

<img width="1275" alt="screen shot 2017-09-16 at 11 43 52 am" src="https://user-images.githubusercontent.com/4000076/30513626-68608c52-9ad4-11e7-9161-cfc873902754.png">
<img width="1223" alt="screen shot 2017-09-16 at 11 43 59 am" src="https://user-images.githubusercontent.com/4000076/30513625-685ef806-9ad4-11e7-981c-2da277616c60.png">

**Comment by tmatsuo** [comment]
The error message suggests that you have `datetime` field for the `publishedDate`.

Regardless of good or bad, our sample code has `VARCHAR` type:
https://github.com/GoogleCloudPlatform/getting-started-php/blob/master/2-structured-data/src/DataModel/Sql.php#L45

I assume that, you already have the table in the database. Did you try the bookshelf sample in other language? If so, maybe drop the table and try it again.

**Comment by jgacuca567** [comment]
@tmatsuo I dropped the database and I am still getting the same error. When you try it are you getting any error?

**Comment by jgacuca567** [comment]
@tmatsuo Any updates, were you able to test?

**Comment by tmatsuo** [comment]
The error message is clear that you have `datetime` field wheres our sample is using `VARCHAR`. Can you show me the output of `describe books;` mysql command?

**Comment by jgacuca567** [comment]
@tmatsuo See attached screenshot. 

<img width="1277" alt="screen shot 2018-01-02 at 7 03 56 pm" src="https://user-images.githubusercontent.com/4000076/34504951-d2e65690-efef-11e7-87d9-9850ca6530b7.png">

**Comment by tmatsuo** [comment]
The source code is creating the VARCHAR field as follows:

```
https://github.com/GoogleCloudPlatform/getting-started-php/blob/master/2-structured-data/src/DataModel/Sql.php#L45
```

Do you have the same version of this file?

**Comment by tmatsuo** [comment]
@jgacuca567

I followed the tutorial, and it worked.

I'm really don't understand you have the `datetime` field for `publishedDate`. Maybe it's easier to have a video conference or sth to debug further. WDYT?

Also, when I tried to add a book, the `Date Published` field was empty (no default value). It was auto completed by your browser maybe?

**Comment by jgacuca567** [comment]
@tmatsuo 
1. I made progress today. Wohoo!!!! 
I was using the old pull requests and after using the new code, I was successfully able to add a book. For the date, what should the right way be because of validation, this is how I posted it. 

I inserted the date as January 3, 2017, shouldn't we use the `DATE` Instead? 

<img width="2560" alt="screen shot 2018-01-02 at 7 08 10 pm" src="https://user-images.githubusercontent.com/4000076/34540628-45095674-f0a3-11e7-9639-206418828ee4.png">


<img width="381" alt="screen shot 2018-01-03 at 4 24 33 pm" src="https://user-images.githubusercontent.com/4000076/34540585-1e5b4604-f0a3-11e7-984b-459bc4f4dd7f.png">
